teamscale-client/src/main/java/com/teamscale/client/PairList.java

```diff
@@ -0,0 +1,312 @@
+/*-------------------------------------------------------------------------+
+|                                                                          |
+| Copyright 2005-2011 The ConQAT Project                                   |
+|                                                                          |
+| Licensed under the Apache License, Version 2.0 (the "License");          |
+| you may not use this file except in compliance with the License.         |
+| You may obtain a copy of the License at                                  |
+|                                                                          |
+|    http://www.apache.org/licenses/LICENSE-2.0                            |
+|                                                                          |
+| Unless required by applicable law or agreed to in writing, software      |
+| distributed under the License is distributed on an "AS IS" BASIS,        |
+| WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. |
+| See the License for the specific language governing permissions and      |
+| limitations under the License.                                           |
++-------------------------------------------------------------------------*/
+package com.teamscale.client;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+/**
+ * A list for storing pairs in a specific order.
+ */
+public class PairList<S, T> implements Serializable, Iterable<Pair<S, T>> {
```

Comment:
Do we really need the PairList? Maybe List<Pair> would be sufficient.

Reply:
Just wanted to make the changes as minimal as possible, but you are right we only need it in one place that can as well be solved with a regular list.